src/openforms/registrations/contrib/stuf_zds/tests/test_backend.py

```diff
@@ -645,6 +660,416 @@ def test_retried_registration_with_internal_reference(self, m, mock_task):
+                "//zkn:object/zkn:heeftAlsInitiator/zkn:gerelateerde/zkn:natuurlijkPersoon/bg:geboortedatum/@stuf:indOnvolledigeDatum": "V",
+                "//zkn:object/zkn:anderZaakObject/zkn:omschrijving": "coordinaat",
+                "//zkn:object/zkn:anderZaakObject/zkn:lokatie/gml:Point/gml:pos": "52.36673378967122 4.893164274470299",
+            },
+        )
+        self.assertXPathNotExists(
+            xml_doc, "//zkn:isVan/zkn:gerelateerde/zkn:omschrijving"
+        )
+        self.assertXPathNotExists(xml_doc, "//zkn:heeft")
+
+        # extraElementen
+        self.assertXPathEquals(
+            xml_doc,
+            "//stuf:extraElementen/stuf:extraElement[@naam='extra']",
+            "BuzzBazz",
+        )
+
+        # don't expect registered data in extraElementen
+        self.assertXPathNotExists(
+            xml_doc, "//stuf:extraElementen/stuf:extraElement[@naam='voornaam']"
+        )
+
+        # PDF report
+        xml_doc = xml_from_request_history(m, 2)
+        self.assertSoapXMLCommon(xml_doc)
+        self.assertXPathEqualDict(
+            xml_doc,
+            {
+                "//zkn:stuurgegevens/stuf:berichtcode": "Di02",
+                "//zkn:stuurgegevens/stuf:functie": "genereerDocumentidentificatie",
+            },
+        )
+
+        xml_doc = xml_from_request_history(m, 3)
+        self.assertSoapXMLCommon(xml_doc)
+        self.assertXPathEqualDict(
+            xml_doc,
+            {
+                "//zkn:object/zkn:inhoud/@stuf:bestandsnaam": "open-forms-inzending.pdf",
+                "//zkn:object/zkn:formaat": "application/pdf",
+            },
+        )
+
+        # attachment
+        xml_doc = xml_from_request_history(m, 4)
+        self.assertSoapXMLCommon(xml_doc)
+        self.assertXPathEqualDict(
+            xml_doc,
+            {
+                "//zkn:stuurgegevens/stuf:berichtcode": "Di02",
+                "//zkn:stuurgegevens/stuf:functie": "genereerDocumentidentificatie",
+            },
+        )
+
+        xml_doc = xml_from_request_history(m, 5)
+        self.assertSoapXMLCommon(xml_doc)
+        self.assertXPathEqualDict(
+            xml_doc,
+            {
+                "//zkn:object/zkn:inhoud/@stuf:bestandsnaam": "my-attachment.doc",
+                "//zkn:object/zkn:formaat": "application/msword",
+            },
+        )
+
+        self.assertEqual(
+            TimelineLogProxy.objects.filter(
+                template="logging/events/stuf_zds_request.txt"
+            ).count(),
+            6,
+        )
+        self.assertEqual(
+            TimelineLogProxy.objects.filter(
+                template="logging/events/stuf_zds_success_response.txt"
+            ).count(),
+            6,
+        )
+
+    @patch("celery.app.task.Task.request")
+    def test_plugin_optional_fields_missing_status_description(self, m, mock_task):
+        submission = SubmissionFactory.from_components(
```

Comment:
I'm missing the variant where `zds_zaaktype_status_omschrijving` is set but `zds_zaaktype_status_code` isn't